### PR TITLE
Remove timeline application 

### PR DESCRIPTION
**Pull request by ananzh**
### Description

In this PR, we made the following changes:

First of all, clean out some advanced settings specific to timeline application and tests.
* Remove timelion:default_rows: This setting defines the default number of rows that a new Timelion sheet should have.
* Remove timelion:default_rows: This setting defines the default number of columns that a new Timelion sheet should have.
* Remove timelion:showTutorial.   

Second, remove src/plugin/timeline completely and modify timeline vis.
Third, remove all the functional tests related to timeline application.

### Issues Resolved
https://github.com/opensearch-project/OpenSearch-Dashboards/issues/3519
https://github.com/opensearch-project/OpenSearch-Dashboards/issues/3593

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by ananzh** [comment]
This PR is blocked by https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3976
But I have a test to integrate the chromedriver fix which shows all test passing: https://github.com/ananzh/OpenSearch-Dashboards/actions/runs/4873744129

**Comment by ananzh** [comment]
manual test on current timeline vis

Screenshot 2023-05-04 at 12 51 06

File: packages/osd-optimizer/limits.yml [hunk 1]
```diff
@@ -74,7 +74,6 @@ pageLoadAssetSize:
   telemetry: 91832
   telemetryManagementSection: 52443
   tileMap: 65337
-  timeline: 29920
```
Comment (joshuarrrr) [inline packages/osd-optimizer/limits.yml:77]: out of curiosity - what does this file even do?

File: cypress/integration/with-security/check_advanced_settings.js [hunk 2]
```diff
@@ -25,13 +25,6 @@ describe('verify the advanced settings are saved', () => {
...
   });
 
-  it('the Timeline default columns field is set to 4', () => {
-    cy.get('[data-test-subj="advancedSetting-editField-timeline:default_columns"]').should(
-      'have.value',
-      4
-    );
-  });
-
   it('the Timeline Maximum buckets field is set to 4', () => {
```
Comment (joshuarrrr) [inline cypress/integration/with-security/check_advanced_settings.js:28]: Should this also be deleted? Or is it used by `vis_type_timeline`?

File: test/functional/config.js [hunk 3]
```diff
@@ -107,9 +106,6 @@ export default async function ({ readConfigFile }) {
       settings: {
         pathname: '/app/management',
       },
-      timeline: {
-        pathname: '/app/timeline',
-      },
```
Comment (joshuarrrr) [inline test/functional/config.js:112]: I'm a little curious how this even worked - as far as I'm aware the actual route to the app is `/app/timelion`

File: src/plugins/vis_type_timeline/server/plugin.ts [hunk 4]
```diff
@@ -193,8 +186,6 @@ export class Plugin {
         schema: schema.string(),
       },
     });
-
-    return deepFreeze({ uiEnabled: config.ui.enabled });
```
Comment (joshuarrrr) [inline src/plugins/vis_type_timeline/server/plugin.ts:197]: Can you explain this change? It would seem out of scope as part of the timeline app removal.

File: src/plugins/vis_type_timeline/server/index.ts [hunk 5]
```diff
@@ -32,41 +32,8 @@ import { PluginConfigDescriptor, PluginInitializerContext } from '../../../core/
...
-    renameFromRoot('vis_type_timeline.graphiteUrls', 'vis_type_timeline.graphiteAllowedUrls'),
-
-    // timelion.ui.enabled deprecation
-    renameFromRoot('timelion.ui.enabled', 'vis_type_timeline.ui.enabled', true),
-    renameFromRoot('vis_type_timelion.ui.enabled', 'vis_type_timeline.ui.enabled', true),
-    renameFromRoot('timeline.ui.enabled', 'vis_type_timeline.ui.enabled', true),
-
-    renameFromRootWithoutMap(
-      'vis_type_timeline.graphiteBlockedIPs',
-      'vis_type_timeline.graphiteDeniedIPs'
```
Comment (joshuarrrr) [inline src/plugins/vis_type_timeline/server/index.ts:67]: Why are we removing these deprecations? I think we still want to map those configs?

File: packages/osd-optimizer/limits.yml (hunk 1 above)
Comment (ananzh) [reply packages/osd-optimizer/limits.yml:77]: I think this limits.yml is a configuration file that is part of the build process, specifically related to the OSD optimizer. This file is used to specify limits for the size of individual plugins in terms of their compiled and bundled JavaScript file sizes. `timeline: 29920` is used for timeline app and `visTypeTimeline: 51933` is for vis_type_timeline. Therefore, we should clean it out.

File: cypress/integration/with-security/check_advanced_settings.js (hunk 2 above)
Comment (ananzh) [reply cypress/integration/with-security/check_advanced_settings.js:28]: `timeline:default_columns` is a specific setting for the timeline application, used to determine the number of default columns shown in the timeline view. In timeline visualization, we don't have such settings.

File: src/plugins/vis_type_timeline/server/plugin.ts (hunk 4 above)
Comment (ananzh) [reply src/plugins/vis_type_timeline/server/plugin.ts:197]: I think my logic here is that since we are removing the timeline application, the `uiEnabled` property and thus the line `deepFreeze({ uiEnabled: config.ui.enabled })` becomes irrelevant and can be safely removed. This is because the timeline visualization doesn't use this property, and it's enabled by default. Only timeline app needs to set `timeline.ui.enabled: true` in the opensearch_dashboards.yml.

File: cypress/integration/with-security/check_advanced_settings.js (hunk 2 above)
Comment (joshuarrrr) [reply cypress/integration/with-security/check_advanced_settings.js:28]: got it, so this was a setting just for timeline sheets.

File: src/plugins/vis_type_timeline/server/plugin.ts (hunk 4 above)
Comment (joshuarrrr) [reply src/plugins/vis_type_timeline/server/plugin.ts:197]: I think I understand like 90% of that. So, the `uiEnabled` property is a setting specific to the timeline app? And therefore irrelevant to `vis_type_timeline`? It just seems a little weird that it was being frozen here if it's not actually used downstream of vis_type_timeline. But I trust you.

File: src/plugins/vis_type_timeline/server/index.ts (hunk 5 above)
Comment (ananzh) [reply src/plugins/vis_type_timeline/server/index.ts]: You are right. I will restore these deprecations. Let's remove them in 3.0.0.

File: src/plugins/vis_type_timeline/server/plugin.ts (hunk 4 above)
Comment (ananzh) [reply src/plugins/vis_type_timeline/server/plugin.ts:197]: Yes, uiEnabled property is a setting specific to the timeline app. In timeline app, there is one line
```
navLinkStatus: visTypeTimeline.isUiEnabled === false ? AppNavLinkStatus.hidden : AppNavLinkStatus.default,
```
that sets the `navLinkStatus` for the Timeline application in OpenSearch Dashboards. This status determines how the navigation link for this application is displayed in the side navigation bar (show up or hidden).

The logic for the `isUiEnabled` value is defined within the `TimelineVisPlugin` class in the `VisTypeTimeline` object.
```
return {
  isUiEnabled: this.initializerContext.config.get().ui.enabled,
};
```
This `this.initializerContext.config.get()` method is used to access the configuration of the timeline app in opensearch_dashboards.yml. So I feel all the logic handling and timeline functions are in vis_type_timeline.

`deepFreeze({ uiEnabled: config.ui.enabled })` is returning an object with a uiEnabled property set to the value of config.ui.enabled, and this object is made immutable by the deepFreeze function. This means that once the PluginSetupContract object is returned by the setup function, the uiEnabled value cannot be changed. I didn't research on how it could be changed without `deepFreeze`. But since there should not be any ui.enabled config left, I just think this line should also be removed. This is my understanding on the whole ui.enabled usage in both timeline app and vis and I hope it would be helpful.

**Comment by joshuarrrr** [comment]
@ananzh There are some failing table vis functional tests. I just started a re-run, but you may want to take a look at the failures.

**Review by joshuarrrr — APPROVED** [review]

**Comment by ananzh** [comment]
cypress fail due to latest merged table tests, which is the same issue as other PRs. not related to timeline. we don't have any timeline app in functional test repo. 
```

       Spec                                              Tests  Passing  Failing  Pending  Skipped  
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ ✔  core-opensearch-dashboards/opensear      01:12        5        5        -        -        - │
  │    ch-dashboards/apps/vis_builder/basi                                                         │
  │    c.spec.js                                                                                   │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  core-opensearch-dashboards/opensear      00:42        4        4        -        -        - │
  │    ch-dashboards/apps/vis_builder/dash                                                         │
  │    board.spec.js                                                                               │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  core-opensearch-dashboards/opensear      00:38        4        4        -        -        - │
  │    ch-dashboards/apps/vis_builder/expe                                                         │
  │    rimental.spec.js                                                                            │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  core-opensearch-dashboards/opensear      00:24        1        1        -        -        - │
  │    ch-dashboards/apps/vis_builder/vis_                                                         │
  │    types/area.spec.js                                                                          │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  core-opensearch-dashboards/opensear      00:25        1        1        -        -        - │
  │    ch-dashboards/apps/vis_builder/vis_                                                         │
  │    types/bar.spec.js                                                                           │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  core-opensearch-dashboards/opensear      00:25        1        1        -        -        - │
  │    ch-dashboards/apps/vis_builder/vis_                                                         │
  │    types/line.spec.js                                                                          │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  core-opensearch-dashboards/opensear      00:23        1        1        -        -        - │
  │    ch-dashboards/apps/vis_builder/vis_                                                         │
  │    types/metric.spec.js                                                                        │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  core-opensearch-dashboards/opensear      00:29        1        1        -        -        - │
  │    ch-dashboards/apps/vis_builder/vis_                                                         │
  │    types/table.spec.js                                                                         │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✖  core-opensearch-dashboards/opensear      01:06        4        -        1        -        3 │
  │    ch-dashboards/apps/vis_type_table/b                                                         │
  │    asic.spec.js                                                                                │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✖  core-opensearch-dashboards/opensear      04:03        9        2        7        -        - │
  │    ch-dashboards/apps/vis_type_table/d                                                         │
  │    ata.spec.js                                                                                 │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✖  core-opensearch-dashboards/opensear      02:35        4        1        3        -        - │
  │    ch-dashboards/apps/vis_type_table/e                                                         │
  │    mbed.spec.js                                                                                │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  core-opensearch-dashboards/opensear      00:56       11       11        -        -        - │
  │    ch-dashboards/apps/vis_type_table/o                                                         │
  │    ptions.spec.js                                                                              │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✖  core-opensearch-dashboards/opensear      02:02        7        1        6        -        - │
  │    ch-dashboards/apps/vis_type_table/s                                                         │
  │    plit.spec.js                                                                                │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  core-opensearch-dashboards/opensear      01:15       34       34        -        -        - │
  │    ch-dashboards/dashboard_sanity_test                                                         │
  │    _spec.js                                                                                    │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  core-opensearch-dashboards/opensear        2ms        -        -        -        -        - │
  │    ch-dashboards/datasource-management                                                         │
  │    -plugin/1_create_datasource.spec.js                                                         │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  core-opensearch-dashboards/opensear        1ms        -        -        -        -        - │
  │    ch-dashboards/datasource-management                                                         │
  │    -plugin/2_datasource_table.spec.js                                                          │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  core-opensearch-dashboards/opensear        1ms        -        -        -        -        - │
  │    ch-dashboards/datasource-management                                                         │
  │    -plugin/3_update_datasource.spec.js                                                         │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
    ✖  4 of 17 failed (24%)                     16:40       87       67       17        -        3  

```

**Review by abbyhu2000 — APPROVED** [review]
